src/ConstructiveSolidGeometry/PointsAndVectors/Points.jl

```diff
@@ -125,10 +125,8 @@ end
 @inline Base.adjoint(pt::CartesianPoint) = CartesianPoint(adjoint(pt.x), adjoint(pt.y), adjoint(pt.z))
 
 # Barycentric combination
-function  Statistics.mean(A::AbstractArray{<:CartesianPoint{T}}; dims = :) where T
-    # Equivalent to A .- Ref(cartesian_zero), but allocation-free:
-    B = reinterpret(CartesianVector{T}, A) # ToDo: Avoid this memory allocation.
-    return cartesian_zero + mean(B; dims = dims)
+function Statistics.mean(A::AbstractArray{<:CartesianPoint}; dims = :)
```

Comment:
I would remove using mean for this completely. It'll just cause confusion if it works for some containers and doesn't work for others.